classes/Cookie.php

```diff
@@ -589,7 +589,7 @@ public function getSession($sessionId)
             $session = new CustomerSession($sessionId);
         }
 
-        if (!empty($session->getId())) {
+        if (Validate::isLoadedObject($session)) {
```

Comment:
@PululuK Are you sure this works? Wouldn't this still throw "undefined variable" ?

Reply:
yup, just tested and I can still replicate the problem in a shop with just the validate change

![image](https://user-images.githubusercontent.com/1352132/120634639-258a4080-c474-11eb-9478-6cfd26028957.png)

```suggestion
        if ($session && Validate::isLoadedObject($session)) {
```

Reply:
PR rebased and squash ;)

Reply:
Thanks guys !